lua/lspconfig/lean3ls.lua

```diff
@@ -6,12 +6,27 @@ configs.lean3ls = {
     cmd = { 'lean-language-server', '--stdio', '--', '-M', '4096', '-T', '100000' },
     filetypes = { 'lean3' },
     root_dir = function(fname)
-      return util.root_pattern 'leanpkg.toml'(fname) or util.find_git_ancestor(fname) or util.path.dirname(fname)
+      -- check if inside stdlib
+      local stdlib_dir
+      local elan_root = util.search_ancestors(fname, function(path)
+        local split = vim.split(path, util.path.sep)
+        if split[#split] == '.elan' then
+          return path
+        end
+      end)
+      if elan_root then
+        local remaining = vim.split(fname:sub(#elan_root + 2), util.path.sep)
+        if remaining[1] == 'toolchains' and remaining[2] then
+          stdlib_dir = util.path.join(elan_root, remaining[1], remaining[2])
+        end
+      end
+
+      return util.root_pattern 'leanpkg.toml'(fname)
+        or stdlib_dir
+        or util.find_git_ancestor(fname)
+        or util.path.dirname(fname)
     end,
     on_new_config = function(config, root)
-      if not util.path.is_file(root .. '/leanpkg.toml') then
```

Comment:
Had this check before assuming the server would only work with a `leanpkg.toml` file next to it, but clearly that isn't the case.